lib/abstract_format.ml

```diff
@@ -909,6 +915,43 @@ and guard_test_of_sf sf : (guard_test_t, err_t) Result.t =
      let%bind rhs = sf_rhs |> guard_test_of_sf |> track ~loc:[%here] in
      GuardTestBinOp {line; op; lhs; rhs} |> return
 
+  (* a record creation : #user{name = "Taro", admin = true} *)
+  | Sf.Tuple (4, [Sf.Atom "record";
+                  Sf.Integer line;
+                  Sf.Atom name;
+                  Sf.List sf_record_fields]) ->
+     let field_of_sf sf =
+       begin match sf with
+       | Sf.Tuple (4, [Sf.Atom "record_field";
+                       Sf.Integer line;
+                       sf_atom_or_wildcard;
+                       sf_guard_test]) ->
+          let%bind field_name = atom_or_wildcard_of_sf sf_atom_or_wildcard in
+          let%bind rhs = guard_test_of_sf sf_guard_test in
+          (line, field_name, rhs) |> return
+       | _ ->
+          Err.create ~loc:[%here] (Err.Not_supported_absform ("cannot reach here", sf)) |> Result.fail
```

Comment:
Could you use `Invalid_input` because there are no patterns except for the above pattern in record creation guard tests?

```suggestion
          Err.create ~loc:[%here] (Err.Invalid_input ("invalid form of record_field", sf)) |> Result.fail
```

Reply:
yes